core/src/test/scala/shapeless/serialization.scala

```diff
@@ -937,11 +943,11 @@ class SerializationTests {
   def testLazy {
     assertSerializable(Lazy(23))
 
-    assertSerializable(implicitly[Lazy[Generic[Wibble]]])
-    assertSerializable(implicitly[Lazy[Generic1[Box, TC1]]])
+    assertSerializableBeforeAfter(implicitly[Lazy[Generic[Wibble]]])(_.value)
+    assertSerializableBeforeAfter(implicitly[Lazy[Generic1[Box, TC1]]])(_.value)
 
```

Comment:
I don't mind adding new tests, but I'm uncomfortable removing existing ones here ... can you explain why this is necessary? If it relates to changes in serialization behaviour due to forcing lazy vals then I think I'd like a deeper understanding of why that's a problem.